File: Task11/CppSamples/sample1.cpp
#include <thread>
#include <iostream>
#include <exception>

int main(void) {
    std::cout << "Thread A started!\n";

    std::thread exceptional_task([](void) -> void {
        std::this_thread::sleep_for(std::chrono::milliseconds(100));
        std::cout << "Thread B started!\n";
        throw std::runtime_error("Thread B aborted!\n");
    });

    for (int i = 0; i < 1000; i++) {
        std::cout << "Thread A doing boring work [-_-]\n";
    }

    std::cout << "Thread A continues\n";

    exceptional_task.join();

    std::cout << "Thread A finished!\n";
}

File: Task11/JavaSamples/Sample1.java
package Task11.JavaSamples;

public class Sample1 {

    public static void main(String[] args) throws Exception {

        var taskB = new Thread(() -> {
            System.out.println("Thread B started!");
            throw new RuntimeException("Thread B aborted!");
        });

        var taskA = new Thread(() -> {                
            taskB.start();
            
            System.out.println("Thread A started!");
            
            for (int i = 0; i < 100; i++) {
                System.out.println("Thread A doing boring work [-_-]");
            }

            try {
                taskB.join();
            } catch (InterruptedException e) {
                throw new RuntimeException("Interrupt in thread A!");
            }
            
            System.out.println("Thread A finished!");
        });

        taskA.start();
        taskA.join();
    }
}
